test/query.test.js

```diff
@@ -4373,4 +4374,28 @@ describe('Query', function() {
 
     assert.strictEqual(books.length, 1);
   });
+
+  it('should avoid sending empty projection to MongoDB server (gh-13065)', async function() {
+    const m = new mongoose.Mongoose();
+    m.set('debug', true);
+
+    const connDebug = m.createConnection(start.uri);
+
+    const schema = new Schema({ name: String });
+    const Test = connDebug.model('Test', schema);
+
+    const consoleInfoStub = sinon.stub(console, 'info').returns();
```

Comment:
I can change the implementation and pass a function to `m.set('debug')`.

By stubbing `findOne` would we be able to check the existence of the `projection` value?
We'd resolve the returned object but we'd still depend on checking the logged message for our test case.
Unless you thought about a different approach.

Reply:
You're right, passing a function to `m.set('debug')` is probably the easiest approach. We should be able to do the same thing by stubbing `findOne()`, but I think the debug approach is easier because we don't need a `sinon.restore()`.

Reply:
I've changed the implementation and passed a function to `m.set('debug')`.
Let me know if you think that further improvements are needed.